src/use-auth.js

```diff
@@ -0,0 +1,68 @@
+import React, {
```

Comment:
Please rename this file to `AuthProvider.js`, the main feature for this file should be the `AuthProvider` not the hook.
And `AuthProvider` should be the default export, it should be the entry point of this file, hooks are just functions to consume the provider.